tests/functional/local-overlay-store/redundant-add-inner.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+set -x
+
+source common.sh
+
+# Avoid store dir being inside sandbox build-dir
+unset NIX_STORE_DIR
+unset NIX_STATE_DIR
+
+storeDirs
+
+initLowerStore
+
+mountOverlayfs
+
+### Do a redundant add
+
+# upper layer should not have it
+expect 1 stat $(toRealPath "$storeBTop/nix/store" "$path")
+
+path=$(nix-store --store "$storeB" --add ../dummy)
+
+# lower store should have it from before
+stat $(toRealPath "$storeA/nix/store" "$path")
+
+# upper layer should still not have it (no redundant copy)
+expect 1 stat $(toRealPath "$storeBTop" "$path")
```

Comment:
Afaiu, "$path" (in the first line) isn't the same as ../dummy, it's the build output of `hermetic.nix`.

It would be better to just inline everything to avoid relying on some hard to track global state

```suggestion
path=$(nix-store --store "$storeA" ../dummy)

expect 1 stat $(toRealPath "$storeBTop/nix/store" "$path")

pathFromB=$(nix-store --store "$storeB" --add ../dummy)

[[ $path == $pathFromB ]]

# lower store should have it from before
stat $(toRealPath "$storeA/nix/store" "$path")

# upper layer should still not have it (no redundant copy)
expect 1 stat $(toRealPath "$storeBTop" "$path")
```